### ISSUE TITLE
const inside switch case statement is a syntax error

### ISSUE THREAD

**Issue by dmnd** [eslint/eslint#1805]
Maybe a bug with ES 6 block bindings?

```
$ eslint -v
v0.14.1


$ eslint -c .eslintrc test.js 

test.js
  4:7  error  Unexpected token const

✖ 1 problem (1 error, 0 warnings)


$ cat test.js 
function fn(foo) {
  switch (foo) {
    case "foo":
      const bar = "bar";
      break;
  }
}


$ cat .eslintrc 
env:
  browser: true,
  node: true

ecmaFeatures:
  blockBindings: true
```

**Comment by nzakas** [comment]
This is a parser error, would you mind filing this at https://github.com/eslint/Espree/issues

**Comment by steelbrain** [comment]
@nzakas I am still getting this error, here's some minimal repro code
```js
module.exports = function(a){
  a.somethin = function(){
    let c = ""
  }
}
```

**Comment by michaelficarra** [comment]
@steelbrain What? Your code doesn't include const or switch.

**Comment by steelbrain** [comment]
@michaelficarra it does include a `let` I was referred to this issue from https://github.com/eslint/eslint/issues/1834

**Comment by michaelficarra** [comment]
The original error was

> Unexpected token const

Is that the error you see? The error was also triggered by interaction with switch. Do you have a switch?

**Comment by steelbrain** [comment]
The error I see is `Unexpected token let`, You can find out more using the repro example above.

**Comment by C0deMaver1ck** [comment]
I just had this error as well.

<img width="1392" alt="screen shot 2015-08-02 at 11 33 59 pm" src="https://cloud.githubusercontent.com/assets/925716/9030314/ffd57d14-396e-11e5-82c2-51d0123d23c3.png">

**Comment by ilyavolodin** [comment]
@C0deMaver1ck Can you share your .eslintrc file please?

**Comment by C0deMaver1ck** [comment]
The previous screenshot was without one. When I used this:

```
env:
  browser: true,
  node: true

ecmaFeatures:
  blockBindings: true
```

I got
```
src/main.js
  3:27  error  Unexpected token >

✖ 1 problem (1 error, 0 warnings)
```

**Comment by IanVS** [comment]
What version are you using?  If you're on `1.0.0`, there is likely an `.eslintrc` file somewhere up the directory structure, or possibly in your `~/`.  Without some kind of eslint configuration, ESLint will do nothing in version `1.0.0`.

**Comment by mysticatea** [comment]
@C0deMaver1ck You are using arrow functions, so need `arrowFunctions: true` in `ecmaFeatures` section. Or...

```
env:
  browser: true
  node: true
  es6: true
```
